idaes/models/unit_models/heat_exchanger_1D.py

```diff
@@ -430,96 +430,125 @@ def build(self):
 
         # Set flow directions for the control volume blocks and specify
         # discretization if not specified.
-        if self.config.flow_type == HeatExchangerFlowPattern.cocurrent:
-            set_direction_hot = FlowDirection.forward
-            set_direction_cold = FlowDirection.forward
+        if self.config.hot_side.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the hot side of the "
+                "heat exchanger. "
+                "Defaulting to finite "
+                "difference method on the hot side."
+            )
+            self.config.hot_side.transformation_method = "dae.finite_difference"
+        if self.config.cold_side.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the cold side of the "
+                "heat exchanger. "
+                "Defaulting to finite "
+                "difference method on the cold side."
+            )
+            self.config.cold_side.transformation_method = "dae.finite_difference"
+
+        if (
+            self.config.hot_side.transformation_method
+            != self.config.cold_side.transformation_method
+        ):
+            raise ConfigurationError(
+                "HeatExchanger1D only supports similar transformation "
+                "methods on the hot and cold side domains for "
+                "both cocurrent and countercurrent flow patterns. "
+                f"Found method {self.config.hot_side.transformation_method} "
+                f"on hot side and method {self.config.cold_side.transformation_method} "
+                "on cold side."
+            )
+        if self.config.hot_side.transformation_method == "dae.collocation":
+            if (
+                self.config.hot_side.transformation_scheme is useDefault
+                or self.config.cold_side.transformation_scheme is useDefault
+            ):
+                raise ConfigurationError(
```

Comment:
Minor comment, but should we have a default here? I vaguely recall that one of the collocation schemes had a bug too.

Reply:
Perhaps. Originally I wanted the user to assume full responsibility for using collocation, but since we need it for energy conservation in upwind schemes I might make `"LAGRANGE-RADAU"` the default. I should test it out on `"LAGRANGE-LEGENDRE"` as well, because that's the one that potentially has a bug in it; it creates additional continuity equations about the values of the function between finite elements, and I can easily see that it fails to do so correctly when flow is moving from 1 to 0.

Reply:
My vague recollection is the it was Lagrange-Legendre that had the possible bug. If so - we should make Lagrange-Radau the default.

Reply:
Is there an issue for this bug with Lagrange-Legendre?

Reply:
@blnicho @andrewlee94 It appears to work just fine for the `HeatExchanger1D` in counter-current mode, and, furthermore, both methods return the same answers for outlet temperatures.